### PR TITLE
Add const qualifier to hipBLAS functions

### PR DESCRIPTION
**Pull request by NaveenElumalaiAMD**
resolves #[LWPMLSE-343](https://ontrack-internal.amd.com/browse/LWPMLSE-343) and issue [#543](https://github.com/ROCmSoftwarePlatform/hipBLAS/issues/543)

Summary of proposed changes:
- Add const qualifier to hipBLAS functions swap, sbmv, spmv, symv, trsm.

### CODE REVIEW

File: library/include/hipblas.h [hunk 1]
```diff
@@ -15281,53 +15291,53 @@ HIPBLAS_EXPORT hipblasStatus_t hipblasStrsmBatched(hipblasHandle_t    handle,
...
+                                                   hipblasSideMode_t   side,
+                                                   hipblasFillMode_t   uplo,
+                                                   hipblasOperation_t  transA,
+                                                   hipblasDiagType_t   diag,
+                                                   int                 m,
+                                                   int                 n,
+                                                   const double*       alpha,
+                                                   const double* const AP[],
+                                                   int                 lda,
+                                                   double*             BP[],
```
Comment (TorreZuk) [inline library/include/hipblas.h:15310]: double* const

File: library/include/hipblas.h [hunk 2]
```diff
@@ -15281,53 +15291,53 @@ HIPBLAS_EXPORT hipblasStatus_t hipblasStrsmBatched(hipblasHandle_t    handle,
...
                                                    hipblasDiagType_t           diag,
                                                    int                         m,
                                                    int                         n,
-                                                   const hipblasDoubleComplex* alpha,
-                                                   hipblasDoubleComplex* const AP[],
+                                                   const hipblasComplex*       alpha,
+                                                   const hipblasComplex* const AP[],
                                                    int                         lda,
-                                                   hipblasDoubleComplex*       BP[],
+                                                   hipblasComplex*             BP[],
```
Comment (TorreZuk) [inline library/include/hipblas.h:15324]: * const

File: library/include/hipblas.h [hunk 3]
```diff
@@ -15281,53 +15291,53 @@ HIPBLAS_EXPORT hipblasStatus_t hipblasStrsmBatched(hipblasHandle_t    handle,
...
+                                                   hipblasSideMode_t                 side,
+                                                   hipblasFillMode_t                 uplo,
+                                                   hipblasOperation_t                transA,
+                                                   hipblasDiagType_t                 diag,
+                                                   int                               m,
+                                                   int                               n,
+                                                   const hipblasDoubleComplex*       alpha,
+                                                   const hipblasDoubleComplex* const AP[],
+                                                   int                               lda,
+                                                   hipblasDoubleComplex*             BP[],
```
Comment (TorreZuk) [inline library/include/hipblas.h:15338]: * const

File: library/src/amd_detail/hipblas.cpp [hunk 4]
```diff
@@ -14204,19 +14214,19 @@ catch(...)
...
+                                    hipblasSideMode_t   side,
+                                    hipblasFillMode_t   uplo,
+                                    hipblasOperation_t  transA,
+                                    hipblasDiagType_t   diag,
+                                    int                 m,
+                                    int                 n,
+                                    const double*       alpha,
+                                    const double* const A[],
+                                    int                 lda,
+                                    double*             B[],
```
Comment (TorreZuk) [inline library/src/amd_detail/hipblas.cpp:14227]: double* const

File: library/src/amd_detail/hipblas.cpp [hunk 5]
```diff
@@ -14239,19 +14249,19 @@ catch(...)
...
+                                    hipblasSideMode_t           side,
+                                    hipblasFillMode_t           uplo,
+                                    hipblasOperation_t          transA,
+                                    hipblasDiagType_t           diag,
+                                    int                         m,
+                                    int                         n,
+                                    const hipblasComplex*       alpha,
+                                    const hipblasComplex* const A[],
+                                    int                         lda,
+                                    hipblasComplex*             B[],
```
Comment (TorreZuk) [inline library/src/amd_detail/hipblas.cpp:14262]: * const

File: library/src/amd_detail/hipblas.cpp [hunk 6]
```diff
@@ -14274,19 +14284,19 @@ catch(...)
...
+                                    hipblasSideMode_t                 side,
+                                    hipblasFillMode_t                 uplo,
+                                    hipblasOperation_t                transA,
+                                    hipblasDiagType_t                 diag,
+                                    int                               m,
+                                    int                               n,
+                                    const hipblasDoubleComplex*       alpha,
+                                    const hipblasDoubleComplex* const A[],
+                                    int                               lda,
+                                    hipblasDoubleComplex*             B[],
```
Comment (TorreZuk) [inline library/src/amd_detail/hipblas.cpp:14297]: * const

**Review by TorreZuk — COMMENTED** [review]
Good to fix these now.  Double check the ones I pointed out

File: library/src/amd_detail/hipblas.cpp [hunk 7]
```diff
@@ -3667,8 +3667,13 @@ catch(...)
...
 // swap_batched
-hipblasStatus_t hipblasSswapBatched(
-    hipblasHandle_t handle, int n, float* x[], int incx, float* y[], int incy, int batchCount)
+hipblasStatus_t hipblasSswapBatched(hipblasHandle_t handle,
+                                    int             n,
+                                    float* const    x[],
+                                    int             incx,
+                                    float* const    y[],
+                                    int             incy,
+                                    int             batchCount)
```
Comment (daineAMD) [inline library/src/amd_detail/hipblas.cpp:3676]: These changes should be made in nvidia_detail/hipblas.cpp as well. Otherwise looks good.

**Review by daineAMD — CHANGES_REQUESTED** [review]

File: library/src/amd_detail/hipblas.cpp (hunk 7 above)
Comment (NaveenElumalaiAMD) [reply library/src/amd_detail/hipblas.cpp:3676]: Made similar changes to nvidia_detail/hipblas.cpp

**Review by daineAMD — APPROVED** [review]
LGTM; I'm not sure of the status of cuda CI, I know it's been flaky somewhat recently. If you don't notice cuda CI passing, you or I can reach out to the CI team, I can also help test manually otherwise.

**Comment by NaveenElumalaiAMD** [comment]
Got an update from Eiden, that he just enabled hipblas on our temporary cuda job, it might take a while to get to this PR http://math-ci.rocm.amd.com/blue/organizations/jenkins/cuda-temporary%2Fprecheckin-cuda%2FhipBLAS/detail/PR-572/1/pipeline

**Pull request merged**